### PR TITLE
Add deferrable option to EmrTerminateJobFlowOperator

### PR DESCRIPTION
**Pull request by syedahsn**
Deferrable mode for `EmrTerminateJobFlowOperator` as well as the associated tests.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -830,11 +849,32 @@ def execute(self, context: Context) -> None:
         self.log.info("Terminating JobFlow %s", self.job_flow_id)
         response = emr.terminate_job_flows(JobFlowIds=[self.job_flow_id])
 
+        if self.deferrable:
```
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/operators/emr.py:909]: oh we're adding deferrable capability to an operator that didn't have a `wait_for_completion` ?

File: airflow/providers/amazon/aws/operators/emr.py [hunk 2]
```diff
@@ -830,11 +849,32 @@ def execute(self, context: Context) -> None:
...
+                    aws_conn_id=self.aws_conn_id,
+                ),
+                method_name="execute_complete",
+                # timeout is set to ensure that if a trigger dies, the timeout does not restart
+                # 60 seconds is added to allow the trigger to exit gracefully (i.e. yield TriggerEvent)
+                timeout=timedelta(seconds=self.waiter_max_attempts * self.waiter_delay + 60),
+            )
+
         if not response["ResponseMetadata"]["HTTPStatusCode"] == 200:
             raise AirflowException(f"JobFlow termination failed: {response}")
```
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/operators/emr.py]: this should be done before the `defer`, right ?

**Review by vandonr-amz — COMMENTED** [review]
&& same remarks as on https://github.com/apache/airflow/pull/31657 for the trigger:
 - no need to cache hook
 - should raise error directly

File: airflow/providers/amazon/aws/operators/emr.py (hunk 1 above)
Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/emr.py:852]: Yea, deferring essentially enforces a `wait_for_completion`. It makes sense to add that functionality in for the synchronous path as well. 
See [this comment](https://github.com/apache/airflow/pull/30864#discussion_r1178385960) in a previous PR

File: airflow/providers/amazon/aws/operators/emr.py (hunk 2 above)
Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/emr.py]: yes good catch.

File: airflow/providers/amazon/aws/operators/emr.py (hunk 1 above)
Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/emr.py:909]: Yea, I don't think there will be any issues with doing something like this.

**Comment by syedahsn** [comment]
Sorry for tagging everyone in the PR. I messed up my commits :sweat_smile:
Everyone is welcome to review though :stuck_out_tongue:

**Review by vandonr-amz — APPROVED** [review]
lessgo

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**